### PR TITLE
feat(progressbar): stacked progressbar

### PR DESCRIPTION
**Pull request by chenyuzhcy**
feature #905 

I'll add tests once we settle on the interface and design.

### CODE REVIEW

File: src/progressbar/progressbar.ts [hunk 1]
```diff
@@ -67,4 +81,32 @@ export class NgbProgressbar {
...
+    return {
+      animated: this.animated, striped: this.striped, showValue: this.showValue, type: this.type, value: this.value,
+          max: this.max
+    }
+  }
 }
+
+@Component({
+  selector: 'ngb-progressbar-stack',
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (maxokorokov) [inline src/progressbar/progressbar.ts]: for more advanced tests `OnPush` change detection won't work

**Comment by maxokorokov** [comment]
Ok, I played with it for some more time. One thing that is not working at all and I've no idea how to make it work is [content projection inside of the template]( https://github.com/ng-bootstrap/ng-bootstrap/pull/1920/files#diff-7b0a94cb195b18937cd50a0254d2db81R24)

Another thing is that's a shame that we're instantiating `ngb-progressbars` and not using them at all afterwards, but they're change detected, etc.

Will add a comment in the issue about this also.

**Comment by pathurs** [comment]
The design should be equivalent to Bootstrap 4 itself.

The two below snippets should show the exact same progress bars.

Bootstrap 4:
```html
<div class="progress">
  <div class="progress-bar bg-success" style="width: 50%"></div>
</div>
```

NG Bootstrap:
```html
<ng-progress [max]="20">
  <ng-progressbar type="success" [value]="10"></ng-progressbar>
</ng-progress>
```

I think the parent should be called "NgbProgress" to conform with the class naming.

NgbProgress should be something like this:

```ts
@Component({
  selector: 'ngb-progress',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
  <div class="progress" [style.height]="height">
    <ng-content></ng-content>
  </div>
  `
})
export class NgbProgress {
  @Input() max: number;
  @Input() animated: boolean;
  @Input() striped: boolean;
  @Input() showValue: boolean;
  @Input() height: string;

  constructor(config: NgbProgressbarConfig) {
    this.max = config.max;
    this.animated = config.animated;
    this.striped = config.striped;
    this.showValue = config.showValue;
    this.height = config.height;
  }
}
```

NgbProgressbar should be something like this:

```ts
@Component({
  selector: 'ngb-progressbar',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
  <div *ngIf="parent; then withParent; else withoutParent"></div>
  <ng-template #withParent>
    <div class="progress-bar{{type ? ' bg-' + type : ''}}{{animated ? ' progress-bar-animated' : ''}}{{striped ?
    ' progress-bar-striped' : ''}}" role="progressbar" [style.width.%]="getPercentValue()"
    [attr.aria-valuenow]="getValue()" aria-valuemin="0" [attr.aria-valuemax]="max">
      <span *ngIf="showValue" i18n="ngb.progressbar.value">{{getPercentValue()}}%</span><ng-content></ng-content>
    </div>
  </ng-template>
  <ng-template #withoutParent>
    <div class="progress" [style.height]="height">
      <div class="progress-bar{{type ? ' bg-' + type : ''}}{{animated ? ' progress-bar-animated' : ''}}{{striped ?
      ' progress-bar-striped' : ''}}" role="progressbar" [style.width.%]="getPercentValue()"
      [attr.aria-valuenow]="getValue()" aria-valuemin="0" [attr.aria-valuemax]="max">
        <span *ngIf="showValue" i18n="ngb.progressbar.value">{{getPercentValue()}}%</span><ng-content></ng-content>
        </div>
      </div>
    </ng-template>
  `
})
export class NgbProgressbar {
  @Input() max: number;
  @Input() animated: boolean;
  @Input() striped: boolean;
  @Input() showValue: boolean;
  @Input() type: string;
  @Input() value = 0;
  @Input() height: string;

  constructor(config: NgbProgressbarConfig, @Optional() parent?: NgbProgress) {
    if (parent) {
      this.max = parent.max;
      this.animated = config.animated == null ? parent.animated : config.animated;
      this.striped = config.striped == null ? parent.striped : config.striped;
      this.showValue = config.showValue == null ? parent.showValue : config.showValue;
      this.height = config.height == null ? parent.height : config.height;
        }
    else {
      this.max = config.max;
      this.animated = config.animated;
      this.striped = config.striped;
      this.showValue = config.showValue;
      this.height = config.height;
    }

    this.type = config.type;
  }

  getValue() {
    return getValueInRange(this.value, this.max);
  }

  getPercentValue() {
    return (100 * this.getValue()) / this.max;
  }
}
```

**Comment by pkozlowski-opensource** [comment]
@chenyuzhcy thnx for the PR but we decided to go with your other suggestion _with_ breaking changes, see: https://github.com/ng-bootstrap/ng-bootstrap/issues/905#issuecomment-410249670. As such we don't intend to merge this PR in the current form. Would be totally awesome if you could prepare another PR based on https://github.com/ng-bootstrap/ng-bootstrap/issues/905#issuecomment-410249670.

@pathurs your suggestion is close to what we intend to implement except that we will go with a breaking change and _always_ require `NgbProgress` directive. See more details in https://github.com/ng-bootstrap/ng-bootstrap/issues/905#issuecomment-410249670

**Pull request closed without merging**